dessia_common/document_generator.py

```diff
@@ -1,44 +1,103 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """ Write document file. """
-from typing import List, Tuple
+from typing import List
 
 import docx
 
+from dessia_common.files import BinaryFile
+
+
+class HeaderFooter:
+    """ Represents a header or footer in a docx document. """
+
+    def __init__(self, text: str, align: str = 'center'):
+        self.text = text
+        self.align = align
+
+    def add_to_section(self, section, is_header):
+        """ Add the header/footer to the document. """
+        if is_header:
+            header = section.header
+        else:
+            footer = section.footer
+        paragraph = header.add_paragraph() if is_header else footer.add_paragraph()
```

Comment:
Use the same variable name for header and footer as they can be used similiarly : 

```
if is_header:
            element = section.header
        else:
            element = section.footer
        paragraph = element.add_paragraph()
```

Reply:
Alternative : 

    def add_to_section(self, section, type_: str = "header"):
        """ Add the header/footer to the document. """
        element = getattr(section, type_)
        paragraph = element.add_paragraph()

Reply:
Yes, I forgot to use getattr, thanks